odbc/src/api/data.rs

```diff
@@ -1633,6 +1754,243 @@ mod unit {
+                (
+                    "2014-11-28 09:23:24",
+                    datetime_expectation_no_millis,
+                    Ok(()),
+                    None,
+                ),
+                (
+                    "2014-11-28 09:23:24.123456789",
+                    datetime_expectation_millis,
+                    Ok(()),
+                    None,
+                ),
+                (
+                    "2014-11-28 09:23:24.1234567890",
+                    datetime_expectation_millis,
+                    Ok(()),
+                    Some(FRACTIONAL_TRUNCATION),
+                ),
+                (
+                    "2014-11-28T09:23:24.123456789Z",
+                    datetime_expectation_millis,
+                    Ok(()),
+                    None,
+                ),
+                (
+                    "2014-11-28T09:23:24.1234567890Z",
+                    datetime_expectation_millis,
+                    Ok(()),
+                    Some(FRACTIONAL_TRUNCATION),
+                ),
+                ("2014-11-28", date_expectation, Ok(()), None),
+                (
+                    "11/28/2014",
+                    datetime_expectation_no_millis,
+                    Err(()),
+                    Some(INVALID_DATETIME_FORMAT),
+                ),
+                (
+                    "2014-35-80",
+                    datetime_expectation_no_millis,
+                    Err(()),
+                    Some(INVALID_DATETIME_FORMAT),
+                ),
+                (
+                    "2014-30-90 09:23:24.1234567890",
+                    datetime_expectation_no_millis,
+                    Err(()),
+                    Some(INVALID_DATETIME_FORMAT),
+                ),
+                (
+                    "30:23:24",
+                    datetime_expectation_no_millis,
+                    Err(()),
+                    Some(INVALID_DATETIME_FORMAT),
+                ),
+            ];
+            test_cases
+                .iter()
+                .for_each(|(input, expectation, result, info)| {
+                    match result {
+                        Ok(_) => {
+                            test_conversion_ok!(
+                                input = Bson::String(input.to_string()),
+                                method = to_datetime,
+                                expected = *expectation,
+                                info = info
+                            );
+                        }
+                        Err(_) => {
+                            test_conversion_err!(
+                                input = Bson::String(input.to_string()),
+                                method = to_datetime,
+                                expected = None,
+                                info = info
+                            );
+                        }
+                    };
+                });
+        }
+
+        #[test]
+        fn string_time_conversions_to_datetimes() {
```

Comment:
The reason I split these off was that for time strings -> datetime, the result is `[todays date] HH:MM:SS[.f]`. Because the date portion is dynamic and using chrono's today func, I thought it might be sufficient just to test the time portion of these. The previous function creates a vec of datetime expectations, so naivetime expectations wouldn't fit.

my thoughts would be:
1. do we think it is worth testing that dynamic date generation + if so any thoughts on the best approach?
2. if not, we might be able to use the macros here too for expectation checking

Reply:
oh I see.
I think that you can simply create the expected values like this:
 ```
       let today_plus_time_no_millis_expectation = Utc::today().and_hms(9, 23, 24);
        let today_plus_time_millis_expectation = Utc::today().and_hms_nano(9, 23, 24, 123456789);
```
and in the previous function add the following to the vector:
```
(
    "09:23:24",
    today_plus_time_no_millis_expectation,
    Ok(()),
    Some(FRACTIONAL_TRUNCATION),
),
(
    "09:23:24.123456789",
    today_plus_time_millis_expectation,
    Ok(()),
    None,
),
```
No?